src/providers/common.ts

```diff
@@ -238,3 +243,20 @@ export interface IRunnerProvider extends ec2.IConnectable, iam.IGrantable {
    */
   getStepFunctionTask(parameters: RunnerRuntimeParameters): stepfunctions.IChainable;
 }
+
+/**
+ * Base class for all providers with common methods used by all providers.
+ *
+ * @internal
+ */
+export abstract class BaseProvider extends Construct {
+  protected labelsFromProperties(defaultLabel: string, propsLabel: string | undefined, propsLabels: string[] | undefined): string[] {
+    if (propsLabels) {
+      return propsLabels;
+    } else if (propsLabel) {
+      return [propsLabel];
+    } else {
+      return [defaultLabel];
+    }
+  }
+}
```

Comment:
No need for `else` after a `return` statement:
```suggestion
    if (propsLabels) {
      return propsLabels;
    }
    if (propsLabel) {
      return [propsLabel];
    }
    return [defaultLabel];
  }
}
```